src/runtime.js

```diff
@@ -67,6 +67,22 @@ setReportError(reportErrorForWin.bind(null, self));
 /** @const @private {string} */
 const TAG = 'runtime';
 
+/**
+ * @typedef {{
+ *  url: string,
+ *  ampdoc: !./service/ampdoc-impl.AmpDocShadow,
+ *  setVisibilityState: (Function|undefined),
```

Comment:
I think these should be `function()`, and eventually changed to the right signature rather than `Function`.